docs/api/index.md

```diff
@@ -159,30 +159,96 @@ When you use `test` in the top level of file, they are collected as part of the
 
 ## Setup and Teardown
 
-These global functions allows you to hook into the life cycle of tests to avoid repeating setup and teardown code. They apply to the current context: the file if they are used at the top-level or the current suite if they are inside a `describe` block.
+These functions allows you to hook into the life cycle of tests to avoid repeating setup and teardown code. They apply to the current context: the file if they are used at the top-level or the current suite if they are inside a `describe` block.
 
 ### beforeEach
 
-- **Type:** `beforeEach(fn: () => Awaitable<void>, timeout: number)`
+- **Type:** `beforeEach(fn: () => Awaitable<void>, timeout?: number)`
 
-  Register a callback to be called before each test in the current context.
+  Register a callback to be called before each of the tests in the current context runs.
+  If the function returns a promise, Vitest waits until the promise resolve before running the test.
+
+  Optionally, you can a timeout (in milliseconds) for specifying how long to wait before terminating. The default is 5 seconds.
+
+  ```ts
+  import { beforeEach } from 'vitest'
+
+  beforeEach(() => {
+    // Clear mocks and add some testing data after before each test run
+    stopMocking().then(() => {
+      addUser({ name: 'Patak '})
+    })
+  })
+
+  // Hook accepting a timeout
+  beforeEach(() => { }, 1000)
```

Comment:
I think we should only include an `async` hook directly, and avoid the second with the timeout (it should be clear from the signature already). And let's use a generic name for the user. Same for other examples
```suggestion
  import { beforeEach } from 'vitest'

  beforeEach(async () => {
    // Clear mocks and add some testing data after before each test run
    await stopMocking()
    addUser({ name: 'John'})
  })
```

Reply:
yeah. This makes more sense